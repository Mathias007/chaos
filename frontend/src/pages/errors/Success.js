import React from "react";
import { Link } from "react-router-dom";

import serverStatuses from "../../_config/serverStatuses";
import styles from "../../styles/styles";

import { Button, Layout, Result } from "antd";
const { Content } = Layout;

const { SUCCESS_STRING } = serverStatuses;

export default function Success(props) {
    const {
        cancelLink,
        continueFunction,
        description,
        message,
        oneButton
    } = props;
    return (
        <Content style={styles.content}>
            <Result
                status={SUCCESS_STRING}
                title={message}
                subTitle={description}
                extra={
                    oneButton ? (
                        <Button key="cancel" onClick={continueFunction}>
                            <Link to={cancelLink}>Zakończ</Link>
                        </Button>
                    ) : (
                        [
                            <Button
                                key="continue"
                                type="primary"
                                onClick={continueFunction}
                            >
                                Kontynuuj
                            </Button>,
                            <Button key="cancel">
                                <Link to={cancelLink}>Zakończ</Link>
                            </Button>
                        ]
                    )
                }
            />
        </Content>
    );
}
