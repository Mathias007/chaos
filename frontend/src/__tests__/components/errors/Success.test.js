import React from "react";
import renderer from "react-test-renderer";
import Success from "../../../pages/errors/Success";

describe("Success test", () => {
    it("Success should match snapshot", () => {
        const component = renderer.create(<Success />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
