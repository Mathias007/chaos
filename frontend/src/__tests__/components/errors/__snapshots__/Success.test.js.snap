// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success test Success should match snapshot 1`] = `
<div
  className="ant-result ant-result-success"
>
  <div
    className="ant-result-icon"
  >
    <i
      aria-label="icon: check-circle"
      className="anticon anticon-check-circle"
    >
      <svg
        aria-hidden="true"
        className=""
        data-icon="check-circle"
        fill="currentColor"
        focusable="false"
        height="1em"
        viewBox="64 64 896 896"
        width="1em"
      >
        <path
          d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm193.5 301.7l-210.6 292a31.8 31.8 0 0 1-51.7 0L318.5 484.9c-3.8-5.3 0-12.7 6.5-12.7h46.9c10.2 0 19.9 4.9 25.9 13.3l71.2 98.8 157.2-218c6-8.3 15.6-13.3 25.9-13.3H699c6.5 0 10.3 7.4 6.5 12.7z"
        />
      </svg>
    </i>
  </div>
  <div
    className="ant-result-title"
  >
    Successfully Purchased Cloud Server ECS!
  </div>
  <div
    className="ant-result-subtitle"
  >
    Order number: 2017182818828182881 Cloud server configuration takes 1-5 minutes, please wait.
  </div>
  <div
    className="ant-result-extra"
  >
    <button
      className="ant-btn ant-btn-primary"
      onClick={[Function]}
      type="button"
    >
      <span>
        Go Console
      </span>
    </button>
    <button
      className="ant-btn"
      onClick={[Function]}
      type="button"
    >
      <span>
        Buy Again
      </span>
    </button>
  </div>
</div>
`;
